src/restconfig/src/main/java/org/geoserver/rest/catalog/DataStoreFileController.java

```diff
@@ -500,9 +499,7 @@ public void dataStorePut(
                         do {
                             ftinfo.setName(originalName + x);
                             x++;
-                        } while (x < 10
-                                && catalog.getFeatureTypeByName(namespace, ftinfo.getName())
-                                        != null);
+                        } while (catalog.getFeatureTypeByName(namespace, ftinfo.getName()) != null);
```

Comment:
I was wondering if keeping a limit, just higher, but the infinite loop would occur by if getFeatureTypeByName is misbehaving, which should break other tests, or take very long if there are millions of layers, but that would cause issues much sooner and in various other places. So +1